crates/radix-engine-toolkit-uniffi/src/builder/transaction_builder.rs

```diff
@@ -212,3 +212,37 @@ impl TransactionBuilderIntentSignaturesStep {
         Ok(notarized_transaction)
     }
 }
+
+#[test]
+#[should_panic(expected = "called `Option::unwrap()` on a `None` value")]
+fn test_sign_with_private_key_panics() {
+    let pub_key = PublicKey::Ed25519 {
+        value: (0..32).collect(),
+    };
+    let prv_key = PrivateKey::new_ed25519((32..64).collect()).unwrap();
+
+    let th = TransactionHeader {
+        network_id: 0,
+        start_epoch_inclusive: 1,
+        end_epoch_exclusive: 2,
+        nonce: 1,
+        notary_public_key: pub_key,
+        notary_is_signatory: false,
+        tip_percentage: 0,
+    };
+
+    let manifest = TransactionManifest::new(
+        Arc::new(Instructions(Vec::new(), 0)),
+        Vec::new(),
+    );
+
+    let intent_sig = TransactionBuilder::new()
+        .header(th)
+        .manifest(manifest)
+        .sign_with_private_key(prv_key.clone());
+    let _intent_sig_other = Arc::clone(&intent_sig);
+
+    // Following call should panic because reference count of intent_sig
```

Comment:
I'm a little worried about the panic here since I think that some users in other languages might want to have multiple references to their transaction builder for a wide array of different reasons. One of them is that they might be doing things one step at a time and kind of misusing the builder.

How about to we do something similar way to the manifest builder (seen [here](https://github.com/radixdlt/radix-engine-toolkit/blob/main/crates/radix-engine-toolkit-uniffi/src/builder/manifest_builder/utils.rs#L20-L27)). We try to unwrap and get the underlying builder or clone to create a new builder if we fail to.